examples/hosts/app-integration/external-data/src/model/taskList.ts

```diff
@@ -349,17 +350,21 @@ export class TaskList extends DataObject<{ InitialState: IBaseDocumentInitialSta
 	};
 
 	protected async initializingFirstTime(props: IBaseDocumentInitialState): Promise<void> {
-		const externalTaskListId = props?.externalTaskListId;
+		const externalTaskListId = props.externalTaskListId;
 		if (externalTaskListId === undefined) {
 			throw new Error(
 				"externalTaskListId not present in instantiation. Cannot instantiate task list",
 			);
 		}
 		this._externalTaskListId = externalTaskListId;
-		// TODO: remove console.log of externalTaskListId once it is used in upcoming PRs.
-		// Linter complains about it not being used and then tried to remove
-		// _externalTaskListId as well. This log is simpy to allow building for now.
-		console.log(this.externalTaskListId);
+		// TODO: The check below is a hack for ensuring that the container is attached
+		// before creating and registering the task list, and importing the task list data
+		// form the external server. This is pretty anti-fluid, so we need a better way to
+		// do this check.
+		if (props.containerUrl === undefined) {
+			throw new Error("container url is not resolved (container is not attached). Cannot add taskList.")
```

Comment:
```suggestion
			throw new Error("containerUrl is required to register the task list for external change notifications")
```
Inside the task list we can't really know why a caller could have failed to provide the container URL, so a more generic error is probably better. The comment is enough to point out the probable cause, IMO.